lib/librato/collector/counter_cache.rb

```diff
@@ -74,22 +75,28 @@ def flush_to(queue, opts={})
       #   increment :foo, :source => user.id
       #
       def increment(counter, options={})
+        metric = counter.to_s
         if options.is_a?(Fixnum)
           # suppport legacy style
           options = {by: options}
         end
         by = options[:by] || 1
-        if options[:source]
-          metric = "#{counter}#{SEPARATOR}#{options[:source]}"
-        else
-          metric = counter.to_s
+        if options[:tags] && options[:tags].respond_to?(:each)
+          options[:tags].sort.each do |k, v|
+            k = k.is_a?(String) ? k.downcase.delete(' ') : k
+            v = v.is_a?(String) ? v.downcase.delete(' ') : v
+            metric = "#{metric}#{SEPARATOR}#{k}#{SEPARATOR}#{v}"
+          end
         end
         if options[:sporadic]
           make_sporadic(metric)
         end
         @lock.synchronize do
-          @cache[metric] ||= 0
-          @cache[metric] += by
+          @cache[metric] = {} unless @cache[metric]
+          @cache[metric][:name] ||= counter.to_s
```

Comment:
I think in this case `metric` already is the same as `counter.to_s`? Is this redundant?